driver-core/src/test/unit/com/mongodb/internal/operation/ClientBulkWriteOperationTest.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.operation;
+
+import com.mongodb.ClusterFixture;
+import com.mongodb.MongoNamespace;
+import com.mongodb.ServerAddress;
+import com.mongodb.WriteConcern;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.bulk.ClientBulkWriteOptions;
+import com.mongodb.client.model.bulk.ClientBulkWriteResult;
+import com.mongodb.client.model.bulk.ClientNamespacedReplaceOneModel;
+import com.mongodb.client.model.bulk.ClientNamespacedWriteModel;
+import com.mongodb.connection.ClusterId;
+import com.mongodb.connection.ConnectionDescription;
+import com.mongodb.connection.ServerConnectionState;
+import com.mongodb.connection.ServerDescription;
+import com.mongodb.connection.ServerId;
+import com.mongodb.connection.ServerType;
+import com.mongodb.internal.binding.ConnectionSource;
+import com.mongodb.internal.binding.ReadWriteBinding;
+import com.mongodb.internal.connection.Connection;
+import com.mongodb.internal.connection.DualMessageSequences;
+import com.mongodb.internal.connection.OperationContext;
+import org.bson.BsonBinaryWriter;
+import org.bson.BsonDocument;
+import org.bson.Document;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.io.BasicOutputBuffer;
+import org.bson.json.JsonReader;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+
+import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+import static com.mongodb.client.model.bulk.ClientReplaceOneOptions.clientReplaceOneOptions;
+import static java.util.Collections.singletonList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ClientBulkWriteOperationTest {
+    private static final MongoNamespace NAMESPACE = new MongoNamespace("testDb.testCol");
+    @Mock(answer = Answers.RETURNS_SMART_NULLS)
```

Comment:
- Let's enable this answer by default, as it is done in https://github.com/mongodb/mongo-hibernate/tree/main/src/test/java/org/mockito/configuration.
- Mark `MongoMockito` as `@Deprecated` and document why it should not be used anymore via `@documented`.